indexer/test/advertisement-walker.test.js

```diff
@@ -282,7 +278,49 @@ describe('processNextAdvertisement', () => {
         head: 'baguqeeraTEST',
         tail: 'baguqeeraTEST',
         lastHead: undefined,
-        status: `Error processing baguqeeraTEST: HTTP request to http://127.0.0.1:${serverPort}/ipni/v1/ad/baguqeeraTEST timed out`
+        status: `Error processing baguqeeraTEST: HTTP request to ${serverUrl}ipni/v1/ad/baguqeeraTEST failed: operation timed out`
+      }
+    })
+  })
+
+  it('skips entries where the server responds with 404 cid not found', async () => {
+    const { serverUrl } = await givenHttpServer(async (req, res) => {
+      if (req.url === `/ipni/v1/ad/${FRISBII_AD_CID}`) {
+        const frisbeeRes = await fetch(FRISBII_ADDRESS + req.url)
+        await assertOkResponse(frisbeeRes)
+        // FIXME: can we pipe `frisbeeRes.body` directly to `res`?
+        // `frisbeeRes.body` is a Web API ReadableStream, `res` is a Node.js WritableStream
```

Comment:
See https://nodejs.org/api/stream.html#streamreadabletowebstreamreadable

<img width="522" alt="Screenshot 2024-09-25 at 09 39 44" src="https://github.com/user-attachments/assets/73566bb1-6859-41d4-955a-55249b57881f">

You can either convert all to WebAPI streams, or all to Node.js streams, then continue as usual

Reply:
Thank you for the pointer!

This seems to work:

```js
import * as stream from 'node:stream'
import { pipeline } from 'node:stream/promises'

// (...)

const frisbiiRes = await fetch(FRISBII_ADDRESS + req.url)
await assertOkResponse(frisbiiRes)
assert(frisbiiRes.body, 'frisbii response does not have any body')
await pipeline(stream.Readable.fromWeb(frisbiiRes.body), res)
```